src/fantasi/Tuura/Fantasi/Main.hs

```diff
@@ -14,13 +17,26 @@ main = do
     let graphMLPath           = optGraphML options
         graphVHDLPath         = optGraphName options
         simulationEnvVhdlPath = optSimName options
+        runAlias              = optAliasHub options
 
-    -- parse graph
-    pangraph <- ((maybe (error "file or graph is malformed") id) . P.parse) <$> readFile graphMLPath
+    when runAlias (print "Running Alias")
+
+    -- read and parse graph, handling error with a message.
+    pangraph <- (`getPangraph` options) <$> readFile graphMLPath
+    
     let graphVHDL   = VHDL.writeGraph pangraph
     let simEnvVHDL  = VHDL.writeEnvironment pangraph
 
     -- output vhdl graph
     writeFile graphVHDLPath graphVHDL
     -- output vhdl simulation environment
     writeFile simulationEnvVhdlPath simEnvVHDL
+
+getPangraph :: ByteString -> Options -> Pangraph
+getPangraph bs options = let
+    applyAlias = if optAliasHub options
+        then aliasHub
+        else Just 
+    maybePangraph = P.parse bs >>= applyAlias
+    in fromMaybe (error errMsg) maybePangraph 
+    where errMsg = "Pangraph is nothing! Does the graph construct?"
```

Comment:
This is not a helpful message: the user of FANTASI has no idea what Pangraph is, and what it means for it to be "nothing". Please make this more informative.

Furthermore, you still seem to be mixing up parsing and aliasing. Why not parse first, report parse errors, and then alias? It makes the code easier to read.

Reply:
You are right, I was thinking as a Pangraph user. I think it I might try to catch Maybe exceptions here.